gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjureJavaLocalCodegenPlugin.java

```diff
@@ -134,4 +150,17 @@ private static void createGenerateTask(
         project.getTasks().named("compileJava").configure(compileJava -> compileJava.dependsOn(generateJava));
         ConjurePlugin.applyDependencyForIdeTasks(project, generateJava.get());
     }
+
+    private static Set<ProductDependency> extractProductDependencies(File irFile) {
+        try {
+            MinimalConjureDefinition conjureDefinition =
+                    OBJECT_MAPPER.readValue(irFile, MinimalConjureDefinition.class);
+            return conjureDefinition
+                    .extensions()
+                    .map(MinimalConjureDefinition.Extensions::productDependencies)
+                    .orElseGet(Collections::emptySet);
+        } catch (IOException e) {
+            throw new SafeRuntimeException("Failed to parse conjure definition", e);
```

Comment:
wasn't the idea that we should be able to limp onwards whatever the block contains?

Reply:
We fail if the block exists and it contains a `recommended-product-dependencies` field which has something thing that doesn't look like product dependencies. I thought it was preferable to fail in this case rather than silently ignore malformed pdeps